zk/l1infotree/tree.go

```diff
@@ -163,11 +176,20 @@ func (mt *L1InfoTree) AddLeaf(index uint32, leaf [32]byte) (common.Hash, error)
 			// the sibling of 0 bit should be the zero hash, since we are in the last node of the tree
 		}
 	}
+
+	mt.allLeaves[leaf] = struct{}{}
+
 	mt.currentRoot = cur
 	mt.count++
 	return cur, nil
 }
 
+func (mt *L1InfoTree) LeafExists(leaf [32]byte) bool {
+	_, ok := mt.allLeaves[leaf]
+	log.Info("All leaves: ", mt.allLeaves)
```

Comment:
you sure it should be Info?

Reply:
Nice catch! This was for debugging 😅 Will delete.